<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="plot_cost"           default="true"/>
    <arg name="plot_input"          default="true"/>
    <arg name="enable_obstacle"     default="true"/>

    <arg name="description_file" default="$(find mppi_panda_raisim)/data/panda_grasp.urdf.xacro"/>

    <param name="robot_description"         command="$(find xacro)/xacro $(arg description_file)
      resources_prefix:='file://'
      add_finger:=true"/>

    <param name="robot_description_raisim"  command="$(find xacro)/xacro $(arg description_file)
      collision_links:=false
      collision_hand:=true
      collision_finger:=true
      collision_format:='obj'
      add_finger:=true
      no_position_limits:=true"/>

    <param name="door_description"         command="$(find xacro)/xacro $(find mppi_panda_raisim)/data/door.urdf"/>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="100"/>
        <param name="use_tf_static"     value="true"/>
        <param name="tf_prefix"         value=""/>
        <remap from="robot_description" to="robot_description"/>
        <remap from="joint_states"      to="/joint_states"/>
    </node>

    <node name="door_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="100"/>
        <remap from="robot_description" to="door_description"/>
        <remap from="joint_states"      to="/door/joint_state"/>
    </node>

    <node name="panda_raisim_control" type="panda_raisim_control" pkg="mppi_panda_raisim" output="screen"
          launch-prefix="gdb -ex run --args">
        <param name="config_file"   value="$(find mppi_panda_raisim)/config/params.yaml"/>
        <rosparam command="load"    file="$(find mppi_panda_raisim)/config/params.yaml"/>
    </node>

    <include file="$(find mppi_panda_raisim)/launch/common.launch">
        <arg name="plot_cost"       value="$(arg plot_cost)"/>
        <arg name="plot_input"      value="$(arg plot_input)"/>
        <arg name="enable_obstacle" value="$(arg enable_obstacle)"/>
    </include>
</launch>
